index.bs

```diff
@@ -5129,6 +5130,56 @@ if the [=authenticator=] does not support [=attestation=].
 : Verification procedure
 :: Return implementation-specific values representing [=attestation type=] [=None=] and an empty [=attestation trust path=].
 
+## Apple Anonymous Attestation Statement Format ## {#sctn-apple-anonymous-attestation}
+
+This attestation statement format is exclusively used by Apple for certain types of Apple devices that support WebAuthn.
+
+: Attestation statement format identifier
+:: apple
+
+: Attestation types supported
+:: [=Anonymous CA=]
+
+: Syntax
+:: The syntax of an Apple attestation statement is defined as follows:
+
+    ```
+    $$attStmtType //= (
+                          fmt: "apple",
+                          attStmt: appleStmtFormat
+                      )
+
+    appleStmtFormat = {
+                          x5c: [ credCert: bytes, * (caCert: bytes) ]
+                      }
+    ```
+
+    The semantics of the above fields are as follows:
+
+    : x5c
+    :: |credCert| followed by its certificate chain, each encoded in X.509 format.
+
+    : credCert
+    :: The credential public key certificate used for attestation, encoded in X.509 format.
+
+: Signing procedure
+::
+    1. Let |authenticatorData| denote the authenticator data for the attestation, and let |clientDataHash| denote the hash of the serialized client data.
+    2. Concatenate |authenticatorData| and |clientDataHash| to form |nonceToHash|.
+    3. Perform SHA-256 hash of |nonceToHash| to produce |nonce|.
+    4. Let Apple anonymous attestation CA generate a certificate for the credential public key and include the |nonce| as an extension with OID ( 1.2.840.113635.100.8.2 ). |credCert| denotes this certificate.
+    5. Set |x5c| to |credCert| followed by its certificate chain.
+
+: Verification procedure
+:: Given the verification procedure inputs |attStmt|, |authenticatorData| and |clientDataHash|, the verification procedure is as follows:
+    1. Verify that |attStmt| is valid CBOR conforming to the syntax defined above and perform CBOR decoding on it to extract the contained fields.
+    2. Verify |x5c| is a valid certificate chain starting from the |credCert| to the Apple WebAuthn root certificate.
+    3. Concatenate |authenticatorData| and |clientDataHash| to form |nonceToHash|.
+    4. Perform SHA-256 hash of |nonceToHash| to produce |nonce|.
+    5. Verify |nonce| matches the value of the extension with OID ( 1.2.840.113635.100.8.2 ) in |credCert|.
```

Comment:
Is there an additional value that needs to be prepended to `nonceToHash` before hashing? In two attestations now from an iPhone X running iOS 14.0.1 there's been a constant `3024a1220420` in the "1.2.840.113635.100.8.2" extension that appears before the hash that matches what I get from hashing `nonceToHash`:

## Attestation 1
**nonce (from hashing authData + clientDataHash):**
```
e6ccfd879f384b14b19a5d633e1ed1d6567d1b40d171b689eddbe7bca5668696
```

**credCert extension value:**
```
3024a1220420e6ccfd879f384b14b19a5d633e1ed1d6567d1b40d171b689eddbe7bca5668696
```

## Attestation 2
**nonce (from hashing authData + clientDataHash):**
```
fbb0475110bdc0ad60245a8672568acab3e9860b1005c7d7aa745976cb7d8ead
```

**credCert extension value:**
```
3024a1220420fbb0475110bdc0ad60245a8672568acab3e9860b1005c7d7aa745976cb7d8ead
```

Reply:
Please disregard, this is operator error. I'm not diving deep enough into the ASN.1 structure to get to the OCTET STRING 😅

Reply:
Good!